GoogleForCreators/web-stories-wp
# Editor Redesign: Dialogs

## Context

All of the editor dialogs, now in updated styles.
And, after discussing with @miina  an idea to consolidate how many times we're specifying button styles in the dialog actions and learning from @ivayloklissarov that the dialog coloration is inverted, I went ahead and updated the background color in the design system for modals and gave the editor and dashboard wrapper components.

## Summary

Changes out the styles of all the dialogs in the editor. 

## Relevant Technical Choices

- Use ds dialog, button, text, input
- Delete old dialog
- If anyone can point me to design specs for the edit media dialog I'm happy to make necessary updates. I couldn't find any in figma so I just made do.
- Creates a `Dialog` wrapper (editor and also dashboard for continuity) so that theme can get inverted. 
- I included the dashboard update for inverting the theme here too because it makes the inversion a lot easier to visualize, at least for me. Sorry for the scope creep. 

## To-do

- [x] The colors here are supposed to be inverted now! In Figma when the dialog was made this wasn't an issue. Made this ticket: https://github.com/google/web-stories-wp/issues/6757 to handle inverting the dialogs and the contents within. 
- The issue for this work #6522 mentions a "story delete" dialog, but I don't think that's a thing?
- The carousel still uses the editor's `Modal` component. When that get's updated the `Modal` can be removed from the editor. 

## User-facing changes

| Dialog | Old | New |
| -- | -- | -- |
| Status Check | <img width="1133" alt="unable to save story - old" src="https://user-images.githubusercontent.com/10720454/110987752-cb725700-832c-11eb-9cc6-61ea3af701de.png"> | ![unable to save story](https://user-images.githubusercontent.com/10720454/111210670-c0b4fd80-858a-11eb-9b9a-23f68f92e441.png) |
| Title Missing | ![missing title -old](https://user-images.githubusercontent.com/10720454/110980260-e5a73780-8322-11eb-97ae-a1cce8e15950.png) | ![missing title](https://user-images.githubusercontent.com/10720454/111210702-cf9bb000-858a-11eb-80ca-a839d4d34464.png) |
| Post Publish | ![newly published - old](https://user-images.githubusercontent.com/10720454/110980272-e6d86480-8322-11eb-8e29-693723f56cee.png) | ![story published](https://user-images.githubusercontent.com/10720454/111210761-de826280-858a-11eb-8ef6-c0fe6c01458e.png) |
| Preview Error |  <img width="433" alt="open preview - old" src="https://user-images.githubusercontent.com/10720454/110987781-d88f4600-832c-11eb-956f-ad0555461d23.png"> | ![open preview](https://user-images.githubusercontent.com/10720454/111210799-ea6e2480-858a-11eb-98e4-7d445523ad27.png) |
| Media Edit | ![edit media - old](https://user-images.githubusercontent.com/10720454/110980266-e63fce00-8322-11eb-9f25-b75d945700bd.png) | ![edit media](https://user-images.githubusercontent.com/10720454/111210837-f528b980-858a-11eb-8e94-ce15706c5768.png) |
| Media Delete | ![delete media - old](https://user-images.githubusercontent.com/10720454/110980261-e5a73780-8322-11eb-873a-8631e3265a8e.png) | ![delete media](https://user-images.githubusercontent.com/10720454/111210858-fe198b00-858a-11eb-88cb-e479b9e2b57a.png) |
| Missing Upload Permissions | <img width="637" alt="access restrictions - old" src="https://user-images.githubusercontent.com/10720454/110987831-e9d85280-832c-11eb-9828-88d791ecbd48.png"> | ![access restrictions](https://user-images.githubusercontent.com/10720454/111210903-0c67a700-858b-11eb-996a-935b97fce0aa.png) |
| 3rd Party Terms | ![3rd party - old](https://user-images.githubusercontent.com/10720454/110980267-e63fce00-8322-11eb-8d71-0c9d7278d93d.png) | ![stock 3rd party](https://user-images.githubusercontent.com/10720454/111210954-17bad280-858b-11eb-96cb-05f4d3edc76e.png) |
| Confirm Page Layout | ![confirm page layout - old](https://user-images.githubusercontent.com/10720454/110980255-e4760a80-8322-11eb-9d0a-238b4906c372.png) | ![confirm page layout](https://user-images.githubusercontent.com/10720454/111210992-243f2b00-858b-11eb-81da-20cfe6fe87d9.png) |
| Video Optimization | <img width="614" alt="video optimization - old" src="https://user-images.githubusercontent.com/10720454/110987864-f5c41480-832c-11eb-8cc9-9de673ef4714.png"> | ![media optimization](https://user-images.githubusercontent.com/10720454/111211042-2ef9c000-858b-11eb-95bc-a226b6903a23.png) |

| Dashboard Dialog | Old | New |
| -- | -- | -- |
| Delete Story | ![Screen Shot 2021-03-15 at 12 40 16 PM](https://user-images.githubusercontent.com/10720454/111211451-a4659080-858b-11eb-88d7-8d6a35931c00.png) | ![delete story - dashboard](https://user-images.githubusercontent.com/10720454/111211468-aaf40800-858b-11eb-8df7-8664e495b035.png) |
| Delete Publisher Logo | ![Screen Shot 2021-03-15 at 12 40 26 PM](https://user-images.githubusercontent.com/10720454/111211496-b47d7000-858b-11eb-9108-0dcade70c5dd.png) | ![remove logo -dashboard](https://user-images.githubusercontent.com/10720454/111211517-ba735100-858b-11eb-8c97-9ad62dc059ee.png) |

## Testing Instructions

Trigger dialogs, like so: 

-  Status check
    - Story must be unable to be saved (triggered this manually in dev)
-  title missing
    - Have a new story, try to publish it without a title 
-  post publish
    - Have a new story, publish it
-  preview error
    - I could only trigger this manually, if someone with more experience using the editor can add input to this?
-  media delete
    - Go to third party media panel, see the horizontal 3 dots, select delete
-  media edit
    - Go to third party media panel, see the horizontal 3 dots, select edit
-  missing upload permission
    - if your user doesn't have upload permissions this will show up
-  video optimization
    - Enable the video optimization experiment and drag in a video to upload
-  3rd party terms
    - Reset your cookies, now load the editor and go to 3rd party media 
-  confirm page layout
    - Have something on your story page, select a page layout from page layouts tab


### QA

See above




This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist




---



Fixes #6522
